src/main/java/org/quiltmc/loader/impl/QuiltLoaderImpl.java

```diff
@@ -172,7 +180,19 @@ public File getConfigDirectory() {
 	}
 
 	public Path getModsDir() {
-		return getGameDir().resolve("mods");
+		if (modDir == null) {
+			// Should not be null ever
+			refreshModDir(gameDir);
+		}
+
+		if (!Files.exists(modDir)) {
+			try {
+				Files.createDirectories(modDir);
+			} catch (IOException e) {
+				throw new RuntimeException("Creating mods directory", e);
```

Comment:
```suggestion
				throw new RuntimeException("Failed to create mods directory", e);
```
I think this should be more descriptive to what the error is

Reply:
This is an exact copy of the error thrown when creating the config folder fails. Both will have to be changed.